sagemaker-python-sdk/mxnet_gluon_cifar10/cifar10.py

```diff
@@ -45,6 +45,9 @@ def train(current_host, hosts, num_cpus, num_gpus, channel_input_dirs, model_dir
 
 
     data_dir = channel_input_dirs['training']
+    CHECKPOINTS_DIR = '/opt/ml/checkpoints'
+    if not os.path.exists(CHECKPOINTS_DIR):
+        os.makedirs(CHECKPOINTS_DIR)
```

Comment:
Is it necessary to make this folder? I thought SageMaker creates it for you.

Reply:
as of now sagemaker does not contain those directories. which is why i am first checking for their existence and then creating them. 
As you suggested, one option could be to get sagemaker to create those directories and use an environment variable like `SM_CHECKPOINTS_DIR` to refer to it. Sagemaker currently does have other environment variables for directories that hold train_data, test_data, model directories and so on. But I was given to understand that we will not be creating new environment variables/directories for the checkpointing folder. 
@sandeep-krishnamurthy we did discuss about this. can you please comment here.

Reply:
Can we include @ishaaq in this conversation as well?

Reply:
Based on the earlier discussion with @ishaaq , it was suggested that, there is no new environment variable or parameters from SageMaker for checkpointing. We can let the users use a directory and recommended being - /opt/ml/checkpoints

SageMaker would be creating this directory post launch, so, this logic is merely to make sure we have that directory.

Reply:
Having said that, MXNet takes the directory as input. So we are open to suggestion by @mvsusp and @ishaaq on which directory to choose and how to fetch that directory for checkpointing.

Reply:
How about we `not` create the dir and only enable checkpoints if it exists?

Reply:
I would suggest to move the CHECKPOINTS_DIR to top of file and get from environment if not set:


```
CHECKPOINTS_DIR = os.environ.get('CHECKPOINTS_DIR', '/opt/ml/checkpoints')
```

Reply:
> How about we **not** create the dir and only enable checkpoints if it exists?

done

Reply:
> I would suggest to move the CHECKPOINTS_DIR to top of file and get from environment if not set:
> 
> ```
> CHECKPOINTS_DIR = os.environ.get('CHECKPOINTS_DIR', '/opt/ml/checkpoints')
> ```

@larroy there was a separate offline discussion to not create new ENV variables